tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -279,34 +280,63 @@ static PyObject * THPVariable_invert(PyObject* self, PyObject* args) {
   END_HANDLE_TH_ERRORS
 }
 
-static Tensor dispatch_to_backend(const Tensor & self, Backend backend) {
+static void lazy_init_cuda() {
+  static std::once_flag once;
+  std::call_once(once, []() {
+    auto module = THPObjectPtr(PyImport_ImportModule("torch.cuda"));
+    if (!module) throw python_error();
+    auto res = THPObjectPtr(PyObject_CallMethod(module.get(), "_lazy_init", ""));
+    if (!res) throw python_error();
+  });
+}
+
+static Tensor dispatch_type(const Tensor & self, const at::Type & type, int device=-1, bool async=false) {
+  if (type.is_cuda()) {
+    lazy_init_cuda();
+  }
   AutoNoGIL no_gil;
-  AutoGPU auto_gpu(self);
-  return self.toBackend(backend);
+  AutoGPU auto_gpu(device != -1 ? device : self.type().is_cuda() ? self.get_device() : -1);
```

Comment:
this is different behavior than before right?

```
>>> torch.randn(2).cuda(1).cuda(-1)

-0.6868
-0.5133
[torch.cuda.FloatTensor of size 2 (GPU 0)]
```

(i.e. before -1 meant 0, but now it means current device).

Reply:
Good catch -- I'll fix the code. It's a little weird because `tensor.cuda()` defaults to the current device, while every other method on tensor (including `.byte()`, `.float()`, etc.) defaults to the tensor's device.